### PR TITLE
build: Remove maintainer-mode related stuff

### PR DESCRIPTION
**Pull request by p-montanus**
As #482 has pointed out, maintainer-mode is abandoned.
Just removing maintainer-mode stuff.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -630,13 +622,8 @@ fi
...
 
-rm -f Makefile
-if test "$enable_maintainer_mode" = yes ; then
-	AC_MSG_NOTICE(creating maintainer Makefile)
-	ln -s maintainer.mak Makefile
-else
-	AC_CONFIG_FILES(Makefile ctags.1)
-fi
+AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([ctags.1])
```
Comment (b4n) [inline configure.ac]: why 2 separate `AC_CONFIG_FILES` calls?

Comment (p-montanus) [inline configure.ac]: Thank you for pointing it out.
To tell the truth, I separated them unconsciously.

I prefer separated autoconf macro calls, the reason is, to simplify autoconf `--trace` outputs.
If not familiar autoconf `--trace` option, try in ctags git working tree:
```
$ autoconf --trace='AH_TEMPLATE:$1'
```
It outputs cpp macro list `#define`-ed or `#undef`-ed in config.h.

If AC_CONFIG_FILES call is unseparated,
`$ autoconf --trace='AC_CONFIG_FILES:$1'` outputs:
```
Makefile ctags.1
```
but, separeated:
```
Makefile
ctags.1
```
the latter is easy to process, a little.

As spearated AC_CONFIG_FILES calls are not popular,
shoud I revert to unseparated call?

Comment (b4n) [inline configure.ac]: I don't mind much.  Actually I would've written
```Autoconf
AC_CONFIG_FILES([Makefile
                 ctags.1])
```
for readability (esp. for packages having quite a few output files) -- FTR `autoconf --trace` outputs the raw argument value so 2 lines, with indentation on the 2nd.

But really this is cosmetic and if you meant to do it for a reason I don't really care (although maybe the next person with low AC knowledge will wonder if there is a behavioral reason for this, maybe).

**Comment by masatake** [comment]
@p-montanus,  thank you for your works related to the build system.
I guess you may have much more patches.
Instead of merging these patches incremental, I would like to add you to our team.
If you have interests, submit a PR which adds your real name, contactable mail address and your background (and/or motivation) about ctgs to docs/developers.rst.

File: configure.ac (hunk 1 above)
Comment (p-montanus) [inline configure.ac]: Thank you for your understanding of my odd reason.

Due to my intension is not to bring wonder to other persons,
I will fix.

**Comment by p-montanus** [comment]
@masatake,
I am pleased to get an invitation to universal-ctags team.
There is something I have to tell you, please allow me to send e-mail.

**Comment by masatake** [comment]
Feel free to contact me.
My mail address is yamato@redhat.com.

**Pull request merged**